paymentsheet/src/main/java/com/stripe/android/paymentsheet/ui/PaymentOptionsUI.kt

```diff
@@ -0,0 +1,269 @@
+package com.stripe.android.paymentsheet.ui
+
+import androidx.compose.foundation.layout.BoxWithConstraints
+import androidx.compose.foundation.layout.PaddingValues
+import androidx.compose.foundation.layout.fillMaxWidth
+import androidx.compose.foundation.lazy.LazyRow
+import androidx.compose.foundation.lazy.items
+import androidx.compose.material.MaterialTheme
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.collectAsState
+import androidx.compose.runtime.getValue
+import androidx.compose.runtime.remember
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.platform.LocalContext
+import androidx.compose.ui.platform.testTag
+import androidx.compose.ui.res.stringResource
+import androidx.compose.ui.unit.Dp
+import androidx.compose.ui.unit.dp
+import com.stripe.android.model.PaymentMethod
+import com.stripe.android.model.PaymentMethodCode
+import com.stripe.android.paymentsheet.PaymentOptionUi
+import com.stripe.android.paymentsheet.PaymentOptionsItem
+import com.stripe.android.paymentsheet.PaymentOptionsState
+import com.stripe.android.paymentsheet.R
+import com.stripe.android.paymentsheet.model.PaymentSelection
+import com.stripe.android.paymentsheet.toPaymentSelection
+import com.stripe.android.paymentsheet.viewmodels.BaseSheetViewModel
+import com.stripe.android.uicore.shouldUseDarkDynamicColor
+import com.stripe.android.uicore.stripeColors
+
+@Composable
+internal fun PaymentOptions(
+    viewModel: BaseSheetViewModel,
+    modifier: Modifier = Modifier,
+) {
+    val context = LocalContext.current
+
+    val state by viewModel.paymentOptionsState.collectAsState()
+    val isEditing by viewModel.editing.collectAsState()
+    val isProcessing by viewModel.processing.collectAsState()
+
+    val nameProvider: (String?) -> String = remember {
+        { code ->
+            val repo = viewModel.lpmResourceRepository.getRepository()
+            val resource = repo.fromCode(code)?.displayNameResource
+            context.getString(resource!!)
```

Comment:
I'm fairly sure this can happen for saved payment methods. Should we just default to a blank string instead of crashing?

Reply:
Actually, we don’t need this anymore. I added a `displayName` property to `SavedPaymentMethod` in https://github.com/stripe/stripe-android/pull/6048 which works the way you describe.